### PR TITLE
[ADF-4973] CLONE - Header - sidenav menu button expanded/collapsed state not exposed via ARIA

### PR DESCRIPTION
**Pull request by mmuller88**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Toolbar button doesn't have aria expanded property

**What is the new behaviour?**
Toolbar button does have aria expanded property

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by mmuller88** [comment]
@pionnegru made me aware that we have a sidenav propery already on aca and I should take that as example https://github.com/Alfresco/alfresco-content-app/blob/development/src/app.config.json#L49

**Comment by mmuller88** [comment]
@eromano Yes I configured the demo shell to take the that from the user preference [initialSidenavExpanded]="expandedSidenav"

expandedSidenav is from the saved user preference.

**Comment by mmuller88** [comment]
I added unit test to test the toggle / attribute behaviour

File: lib/core/layout/components/header/header.component.ts [hunk 1]
```diff
@@ -49,18 +49,29 @@ export class HeaderLayoutComponent implements OnInit {
     @Input() showSidenavToggle: boolean = true;
 
     /** Emitted when the sidenav button is clicked. */
-    @Output() clicked = new EventEmitter<any>();
+    @Output() clicked = new EventEmitter<boolean>();
+
+    @Input() initialSidenavExpanded: boolean = true;
+
+    sidenavExpanded: boolean = true;
```
Comment (DenysVuika) [inline lib/core/layout/components/header/header.component.ts]: Add the @Input attribute to allow binding from the Layout
```suggestion
    @Input() sidenavExpanded: boolean = true;
```

File: lib/core/layout/components/header/header.component.ts [hunk 2]
```diff
@@ -49,18 +49,29 @@ export class HeaderLayoutComponent implements OnInit {
     @Input() showSidenavToggle: boolean = true;
 
     /** Emitted when the sidenav button is clicked. */
-    @Output() clicked = new EventEmitter<any>();
+    @Output() clicked = new EventEmitter<boolean>();
+
+    @Input() initialSidenavExpanded: boolean = true;
```
Comment (DenysVuika) [inline lib/core/layout/components/header/header.component.ts]: remove this in favor of the existing `sidenavExpanded` with extra @Input()

File: lib/core/layout/components/header/header.component.html [hunk 3]
```diff
@@ -43,6 +44,7 @@
         class="mat-icon-button adf-menu-icon"
         mat-icon-button
         (click)="toggleMenu()"
+        [attr.aria-expanded]="isSidenavExpanded()"
```
Comment (DenysVuika) [inline lib/core/layout/components/header/header.component.html]:
```suggestion
        [attr.aria-expanded]="sidenavExpanded"
```

File: lib/core/layout/components/header/header.component.html [hunk 4]
```diff
@@ -10,6 +10,7 @@
...
```
Comment (DenysVuika) [inline lib/core/layout/components/header/header.component.html]:
```suggestion
        [attr.aria-expanded]="sidenavExpanded"
```

File: docs/core/components/header.component.md [hunk 5]
```diff
@@ -43,6 +43,7 @@ body of the element:
 | position | `string` | "start" | The side of the page that the drawer is attached to (can be 'start' or 'end') |
 | redirectUrl | `string \| any[]` | "/" | The router link for the application logo, when clicked. |
 | showSidenavToggle | `boolean` | true | Toggles whether the sidenav button will be displayed in the header or not. |
+| initialSidenavExpanded | `boolean` | true | Toggles whether the sidenav is initial expanded or not. |
```
Comment (DenysVuika) [inline docs/core/components/header.component.md]: not needed if we have @Input for the `sidenavExpanded`

File: demo-shell/src/app/components/app-layout/app-layout.component.html [hunk 6]
```diff
@@ -5,8 +5,8 @@
         <ng-template let-toggleMenu="toggleMenu">
 
             <adf-layout-header id="adf-header" [title]="title | translate" [redirectUrl]="redirectUrl" [logo]="logo"
-                [tooltip]="tooltip | translate" [showSidenavToggle]="showMenu" [color]="color" [position]="position"
-                (clicked)=toggleMenu($event)>
+                [tooltip]="tooltip | translate" [showSidenavToggle]="showMenu" [initialSidenavExpanded]="expandedSidenav"
```
Comment (DenysVuika) [inline demo-shell/src/app/components/app-layout/app-layout.component.html]:
```suggestion
                [tooltip]="tooltip | translate" [showSidenavToggle]="showMenu" [sidenavExpanded]="expandedSidenav"
```

File: lib/core/layout/components/header/header.component.ts [hunk 7]
```diff
@@ -49,18 +49,29 @@ export class HeaderLayoutComponent implements OnInit {
...
 
     /** The side of the page that the drawer is attached to (can be 'start' or 'end') */
     @Input() position = 'start';
 
     toggleMenu() {
         this.clicked.emit(true);
+        this.sidenavExpanded = !this.sidenavExpanded;
+    }
+
+    isSidenavExpanded(): boolean {
```
Comment (DenysVuika) [inline lib/core/layout/components/header/header.component.ts]: not needed as `sidenavExpanded` is already public and can be accessed directly

**Pull request merged**